core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -114,68 +125,69 @@ public TenantObject setEnabled(final boolean flag) {
+        return subjectdns == null ? null : subjectdns.get(0);
     }
 
     /**
-     * Sets the trusted certificate authority to use for authenticating
-     * devices of this tenant.
+     * Adds a trust anchor to this tenant's trust store using the given public key and subject DN.
+     * 
+     * @param publicKey The trusted CA's public key.
+     * @param subjectDn The trusted CA's subject DN.
      * 
-     * @param publicKey The CA's public key.
-     * @param subjectDn The CA's subject DN.
      * @return This tenant for command chaining.
      * @throws NullPointerException if any of the parameters is {@code null}.
      */
     @JsonIgnore
-    public TenantObject setTrustAnchor(final PublicKey publicKey, final X500Principal subjectDn) {
+    public TenantObject addTrustAnchor(final PublicKey publicKey, final X500Principal subjectDn) {
 
         Objects.requireNonNull(publicKey);
         Objects.requireNonNull(subjectDn);
 
-        final JsonObject trustedCa = new JsonObject();
-        trustedCa.put(TenantConstants.FIELD_PAYLOAD_SUBJECT_DN, subjectDn.getName(X500Principal.RFC2253));
-        trustedCa.put(TenantConstants.FIELD_PAYLOAD_PUBLIC_KEY, publicKey.getEncoded());
-        return setProperty(TenantConstants.FIELD_PAYLOAD_TRUSTED_CA, trustedCa);
+        final JsonObject trustedCa = new JsonObject()
+                .put(TenantConstants.FIELD_PAYLOAD_SUBJECT_DN, subjectDn.getName(X500Principal.RFC2253))
+                .put(TenantConstants.FIELD_PAYLOAD_PUBLIC_KEY, publicKey.getEncoded());
+
+        return addTrustedCaConfiguration(trustedCa);
     }
 
     /**
-     * Sets the trusted certificate authority to use for authenticating
-     * devices of this tenant.
+     * Adds a trusted certificate authority to this tenant's trust store.
+     * <p>
+     * The certificate is use to authenticate devices of this tenant.
      * 
      * @param certificate The CA certificate.
      * @return This tenant for command chaining.
      * @throws NullPointerException if certificate is {@code null}.
      * @throws IllegalArgumentException if the certificate cannot be (binary) encoded.
      */
     @JsonIgnore
-    public TenantObject setTrustAnchor(final X509Certificate certificate) {
+    public TenantObject addTrustAnchor(final X509Certificate certificate) {
 
         Objects.requireNonNull(certificate);
 
         try {
             final JsonObject trustedCa = new JsonObject()
+                    .put(TenantConstants.FIELD_PAYLOAD_SUBJECT_DN, certificate.getSubjectX500Principal().getName())
                     .put(TenantConstants.FIELD_PAYLOAD_CERT, certificate.getEncoded());
-            return setProperty(TenantConstants.FIELD_PAYLOAD_TRUSTED_CA, trustedCa);
-        } catch (CertificateEncodingException e) {
+            return addTrustedCaConfiguration(trustedCa);
+        } catch (final CertificateEncodingException e) {
```

Comment:
this method should probably just delegate to *addTrustAnchor(PublicKey, X500Principal)*, shouldn't it?